server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -406,11 +409,10 @@ DateTimeZone rewriteTimeZone(QueryShardContext context) throws IOException {
                     // rounding rounds down, so 'nextTransition' is a good upper bound
                     final long high = nextTransition;
 
-                    final DocValueFormat format = ft.docValueFormat(null, null);
-                    final Object formattedLow = format.format(low);
-                    final Object formattedHigh = format.format(high);
+                    final Object formattedLow = DocValueFormat.RAW.format(low);
```

Comment:
Since the field is a date, it's a bit confusing to use the RAW format imo, maybe use a `new DocValueFormat.Date(epoch_millis, DateTimeZone.UTC)` instead?

Reply:
I was even thinking about just boxing the long to a Long because there really isn't much more to it in this case. Would that work?

Reply:
It relies on the assumption that we store millis internally, but I think that's fine. :+1: